doubly_linked_list.py

```diff
@@ -3,58 +3,107 @@
+            tail.prev_node = new_node
+
+            self.node_count += 1
+            return True
         else:
-            tail = self.fetch_tail()
-            tail.next_node = Node(key=None, value=value, next_node=None, prev_node=tail)
-        return self.head
+            new_node = Node(key=None, value=value, next_node=None, prev_node=tail)
 
-    def delete_node(self, value):
+            new_node.prev_node = tail.prev_node
+            tail.prev_node.next_node = new_node
+            new_node.next_node = tail
+            tail.prev_node = new_node
+            self.node_count += 1
+            return True
+
+    def delete(self, value):
         temp = self.fetch_head()
-        while temp.next_node and temp:
+        #[1]
+        while temp:
             if temp.value == value:
-                temp.prev_node.next_node = temp.next_node
-                temp.next_node.prev_node = temp.prev_node
-                break
+                if temp.prev_node:
+                    temp.prev_node.next_node = temp.next_node
+                if temp.next_node:
+                    temp.next_node.prev_node = temp.prev_node
+                self.node_count -= 1
+                return True
             temp = temp.next_node
-        return self.fetch_head()
+        return False
 
     def print_doubly_linked_list(self):
         temp = self.fetch_head()
-        while temp:
+        temp = temp.next_node
+        while temp.value:
             print(temp.value, '<->', end='')
             temp = temp.next_node
         print('None')
 
+    def get_position(self, value):
+        position = 0
+        temp = self.fetch_head()
+        while temp:
+            position += 1
+            if temp.value == value:
+                return position
+            temp = temp.next_node
+        return -1
+
+    def get_element(self, position):
+        value = None
+        if position > self.node_count or position <= 0:
+            return value
+        temp = self.fetch_head()
+        count = 1
+        while temp:
+            if position == count:
+                return temp.value
+            count += 1
+            temp = temp.next_node
+        return value
+
+
 def main():
     # Creating a Doubly Linked List
     doubly_linked_list = DoublyLinkedList()
 
     # Inserting nodes into the linked list
-    doubly_linked_list.add_node(10)
-    doubly_linked_list.add_node(20)
-    doubly_linked_list.add_node(30)
+    doubly_linked_list.add(10)
+    doubly_linked_list.add(20)
+    doubly_linked_list.add(30)
 
     # Displaying the initial state of the linked list
     print("Doubly Linked List:")
     doubly_linked_list.print_doubly_linked_list()
 
     # Deleting a node from the linked list
     value_to_delete = 20
-    doubly_linked_list.delete_node(value_to_delete)
+    doubly_linked_list.delete(value_to_delete)
 
     # Displaying the linked list after deletion
     print(f"\nDoubly Linked List after deleting node with value {value_to_delete}:")
```

Comment:
Missing tests code for all methods! Add for missing methods POSITIVE and NEGATIVE cases. use CHATGPT

Reply:
Added more cases

Reply:
Where is method for `get_element` and `get_position` ??? Both positive and negative??